aea/cli/publish.py

```diff
@@ -139,33 +152,185 @@ def _check_is_item_in_local_registry(
         )
     except click.ClickException as e:
         raise click.ClickException(
-            f"Dependency is missing. {str(e)}\nPlease push it first and then retry."
+            f"Dependency is missing. {str(e)}\nPlease push it first and then retry or use {PUSH_ITEMS_FLAG} flag to push automatically"
         )
 
 
-def _save_agent_locally(ctx: Context, is_mixed: bool = False) -> None:
+class BaseRegistry(ABC):
```

Comment:
this is very useful to have. Maybe move to another Python module like `aea.cli.registry.base`. OR, also consider including it into AEA core, e.g. `aea.registry` (note: `aea.registries.*` is just for internal use, not related to AEA registry)

Reply:
The reason is, in future refactorings, we surely want to use this class for registry-related features.

Reply:
the same applies below for implementations of this interface

Reply:
Yes, although please keep it in CLI for now; let's avoid doing large changes during minors.